### PR TITLE
hdf5: update to use parallel version of hdf5

### PR DESCRIPTION
**Pull request by abhinavsns**
hdf5 parallel is required for certain applications and should not conflict with serial applications as they are part of the same source code. Having both creates linking conflicts.
Please revert PR https://github.com/Homebrew/homebrew-core/pull/167477 and remove the formula hdf5-mpi as it becomes a duplicate of this. 

- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `HOMEBREW_NO_INSTALL_FROM_API=1 brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

File: Formula/h/hdf5.rb [hunk 1]
```diff
@@ -7,63 +7,66 @@ class Hdf5 < Formula
   revision 1
   version_scheme 1
 
-  # This regex isn't matching filenames within href attributes (as we normally
-  # do on HTML pages) because this page uses JavaScript to handle the download
-  # buttons and the HTML doesn't contain the related URLs.
   livecheck do
-    url "https://www.hdfgroup.org/downloads/hdf5/source-code/"
-    regex(/>\s*hdf5[._-]v?(\d+(?:\.\d+)+)(?:-\d+)?\.t/i)
+    formula "hdf5"
```
Comment (SMillerDev) [inline Formula/h/hdf5.rb:11]: It can't use itself to livecheck. Please keep the original

File: Formula/h/hdf5.rb [hunk 2]
```diff
@@ -7,63 +7,66 @@ class Hdf5 < Formula
...
   end
 
   bottle do
```
Comment (SMillerDev) [inline Formula/h/hdf5.rb:14]: Don't modify this block please

File: Formula/h/hdf5.rb [hunk 3]
```diff
@@ -7,63 +7,66 @@ class Hdf5 < Formula
...
+      F90=mpif90
     ]
-    inreplace_cxx_files << "build/src/libhdf5.settings" if OS.linux?
-    inreplace inreplace_cxx_files, Superenv.shims_path/ENV.cxx, ENV.cxx
+    args << "--with-zlib=#{Formula["zlib"].opt_prefix}" if OS.linux?
 
-    system "cmake", "--build", "build"
-    system "cmake", "--install", "build"
+    system "./configure", *args
+    system "make", "install"
```
Comment (SMillerDev) [inline Formula/h/hdf5.rb:69]: We don't want to get rid of CMake since upstream recommends CMake

**Review by SMillerDev — CHANGES_REQUESTED** [review]
There are some pretty big changes here that the description does not explain the reasons for. Please only change the serial vs parallel part in this review

**Comment by abhinavsns** [comment]
Okay! I am starting to see the issues here. Hdf5 cannot be compiled with both libcpp-hdf5 (cpp bindings) and parallel option. Probably that is why hdf5-mpi exists. Maybe it is best to introduce petsc-mpi and petsc-complex-mpi for hdf5-mpi version? I am new to brew and unsure how to resolve this issue. Some help would be appreciated. @chenrui333 @fxcoudert

File: Formula/h/hdf5.rb (hunk 1 above)
Comment (abhinavsns) [reply Formula/h/hdf5.rb]: Done!

File: Formula/h/hdf5.rb (hunk 2 above)
Comment (abhinavsns) [reply Formula/h/hdf5.rb]: Done

File: Formula/h/hdf5.rb (hunk 3 above)
Comment (abhinavsns) [reply Formula/h/hdf5.rb]: Done

**Comment by iMichka** [comment]
`petsc` has 262 downloads in the last 30 days; that's not much. And I am not sure how much success the two new ones would have.

Maybe this could be hosted in a tap: https://docs.brew.sh/Taps, https://docs.brew.sh/How-to-Create-and-Maintain-a-Tap

**Comment by cho-m** [comment]
Going to close this PR as current change is not the right approach. It doesn't package the standard `hdf5` and just creates a duplicate of `hdf5-mpi`.

`hdf5` and `hdf5-mpi` should remain separate formulae, similar to how they are distributed by major Linux repositories (e.g. Debian, Fedora, Arch Linux, etc).

---

One thing you can explore is whether `hdf5-mpi` can be made keg-only.

Another alternative is to wait for new conflict handling (https://github.com/Homebrew/brew/issues/16398), which would automatically unlink one of the formulae rather than hard blocking on conflicts. Behavior would be similar to manual approach to conflicts (i.e. `brew unlink <conflicting-formula>` should allow you to install the other formula)

**Pull request closed without merging**